diff --git a/protocol/group/urltest.go b/protocol/group/urltest.go
index 8b28810..8bd85b3 100644
--- a/protocol/group/urltest.go
+++ b/protocol/group/urltest.go
@@ -366,7 +366,7 @@ func (g *URLTestGroup) urlTest(ctx context.Context, force bool) (map[string]uint
 		return result, nil
 	}
 	defer g.checking.Store(false)
-	b, _ := batch.New(ctx, batch.WithConcurrencyNum[any](10))
+	b, _ := batch.New(ctx, batch.WithConcurrencyNum[any](20))
 	checked := make(map[string]bool)
 	var resultAccess sync.Mutex
 	for _, detour := range g.outbounds {
